### PR TITLE
Fix version flag output

### LINKED ISSUE #1169: Pelican `--version` prints to `stderr` instead of `stdout`
Pelican Version:
_Run `./pelican --version` to get version number._

Pelican Service:
- [ ] Client
- [ ] Plugin
- [ ] Registry
- [ ] Director
- [ ] Origin
- [ ] Cache
- [ ] Other (please give the detail)

**Describe the bug**
_A clear and concise description of what the bug is._

As reported by @matyasselmeci,  Pelican `--version` was printed out stdout, but currently it prints to stderr

**To Reproduce**
_Steps to reproduce the behavior._

**Expected behavior**
_A clear and concise description of what you expected to happen._

**Screenshots**
_If applicable, add screenshots to help explain your problem._

**Desktop (please complete the following information):**
 - OS: [e.g. Linux, MacOS]
 - Browser: [e.g. chrome, safari, if the bug relates to any web user interface]

**Additional context**
_Add any other context about the problem here._

### PR DESCRIPTION
**Pull request by haoming29**
Fixes #1169

### CODE REVIEW

**Comment by haoming29** [comment]
The macOS test failed at installing dependencies, which I have no clue but shouldn't be related to this PR.

**Comment by haoming29** [comment]
> Is the version_test.sh script being run anywhere? I'd suggest adding it to the GitHub actions test action.

Good call! I missed that part

**Comment by haoming29** [comment]
Since Emma is left for the day and we want to release the patch ASAP, assigned @jhiemstrawisc to take another look

File: cmd/main_test.go [hunk 1]
```diff
@@ -110,37 +110,20 @@ func TestHandleCLIVersionFlag(t *testing.T) {
 	batchTest := func(t *testing.T, arguments []string, expected string) {
 		got := ""
 
-		if expected == mockVersionOutput {
-			// Redirect output to a pip
-			oldStderr := os.Stderr
-			r, w, _ := os.Pipe()
-			os.Stderr = w
+		// Redirect output to a pip
```
Comment (jhiemstrawisc) [inline cmd/main_test.go:113]: pip --> pipe

File: cmd/main_test.go [hunk 2]
```diff
@@ -110,37 +110,20 @@ func TestHandleCLIVersionFlag(t *testing.T) {
...
-			err := handleCLI(arguments)
-			require.NoError(t, err)
+		err := handleCLI(arguments)
+		require.NoError(t, err)
 
-			// Close the write of pip and redirect output back to Stderr
-			w.Close()
-			out, _ := io.ReadAll(r)
-			os.Stderr = oldStderr
+		// Close the write of pip and redirect output back to Stderr
```
Comment (jhiemstrawisc) [inline cmd/main_test.go:121]: pip --> pipe

File: github_scripts/version_test.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+# This tests the --version flag of various Pelican binaries (pelican/stash/osdf)
+
+set -e
+
+mkdir -p /tmp/pelican-test/version-test
+
+export PELICAN_CONFIGDIR=/tmp/pelican-test/version-test
+export OSDF_CONFIGDIR=/tmp/pelican-test/version-test
+export STASH_CONFIGDIR=/tmp/pelican-test/version-test
```
Comment (jhiemstrawisc) [inline github_scripts/version_test.sh:26]: With the merge of #1162, let's avoid setting the now-extraneous OSDF/STASH env vars

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]
Minor edits. The guts all look good to me.

**Review by jhiemstrawisc — APPROVED** [review]
LGTM

**Comment by jhiemstrawisc** [comment]
Since Emma is out and this is urgent, I'm going to merge.

**Pull request merged**